src/modules/sensors/vehicle_angular_velocity/VehicleAngularVelocity.cpp

```diff
@@ -0,0 +1,226 @@
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "VehicleAngularVelocity.hpp"
+
+#include <px4_log.h>
+
+using namespace time_literals;
+using namespace matrix;
+
+VehicleAngularVelocity::VehicleAngularVelocity() :
+	ModuleParams(nullptr),
+	WorkItem(px4::wq_configurations::rate_ctrl),
+	_cycle_perf(perf_alloc(PC_ELAPSED, "vehicle_angular_velocity: cycle time")),
+	_interval_perf(perf_alloc(PC_INTERVAL, "vehicle_angular_velocity: interval")),
+	_sensor_gyro_latency_perf(perf_alloc(PC_ELAPSED, "vehicle_angular_velocity: sensor gyro latency"))
+{
+}
+
+VehicleAngularVelocity::~VehicleAngularVelocity()
+{
+	Stop();
+
+	perf_free(_cycle_perf);
+	perf_free(_interval_perf);
+	perf_free(_sensor_gyro_latency_perf);
+}
+
+bool
+VehicleAngularVelocity::Start()
+{
+	// initialize thermal corrections as we might not immediately get a topic update (only non-zero values)
+	_scale = Vector3f{1.0f, 1.0f, 1.0f};
+	_offset.zero();
+	_bias.zero();
+
+	// force initial updates
+	ParametersUpdate(true);
+	SensorBiasUpdate(true);
+
+	_sensor_correction_sub.register_callback();
+
+	return SensorCorrectionsUpdate(true);
+}
+
+void
+VehicleAngularVelocity::Stop()
+{
+	Deinit();
+
+	// clear all registered callbacks
+	for (auto sub : _sensor_gyro_sub) {
+		sub.unregister_callback();
+	}
+
+	_sensor_correction_sub.unregister_callback();
+}
+
+void
+VehicleAngularVelocity::SensorBiasUpdate(bool force)
+{
+	if (_sensor_bias_sub.updated() || force) {
+		sensor_bias_s bias;
+
+		if (_sensor_bias_sub.copy(&bias)) {
+			// TODO: should be checking device ID
+			_bias(0) = bias.gyro_x_bias;
+			_bias(1) = bias.gyro_y_bias;
+			_bias(2) = bias.gyro_z_bias;
+		}
+	}
+}
+
+bool
+VehicleAngularVelocity::SensorCorrectionsUpdate(bool force)
+{
+	// check if the selected gyro has updated
+	if (_sensor_correction_sub.updated() || force) {
+
+		sensor_correction_s corrections{};
+		_sensor_correction_sub.copy(&corrections);
+
+		// TODO: should be checking device ID
+		if (_selected_gyro == 0) {
+			_offset = Vector3f{corrections.gyro_offset_0};
+			_scale = Vector3f{corrections.gyro_scale_0};
+
+		} else if (_selected_gyro == 1) {
+			_offset = Vector3f{corrections.gyro_offset_1};
+			_scale = Vector3f{corrections.gyro_scale_1};
+
+		} else if (_selected_gyro == 2) {
+			_offset = Vector3f{corrections.gyro_offset_2};
+			_scale = Vector3f{corrections.gyro_scale_2};
+
+		} else {
+			_offset = Vector3f{0.0f, 0.0f, 0.0f};
+			_scale = Vector3f{1.0f, 1.0f, 1.0f};
+		}
+
+		// update the latest gyro selection
+		if (_selected_gyro != corrections.selected_gyro_instance) {
+			if (corrections.selected_gyro_instance < MAX_GYRO_COUNT) {
+				// clear all registered callbacks
+				for (auto sub : _sensor_gyro_sub) {
```

Comment:
for (auto& sub : _sensor_gyro_sub) {

Reply:
Ouch, fixed. https://github.com/PX4/Firmware/pull/12662

Thanks @bkueng